app/blueprints/marketing.py

```
@@ -0,0 +1,390 @@
+        return False
+
+
+def _serve_lead_magnet_file(magnet: LeadMagnet, magnet_name: str):
+    """Return a Flask response for the magnet file or raise OSError if not found."""
+    safe_dir = current_app.config.get('LEAD_MAGNET_DIR', os.path.join(current_app.root_path, 'static', 'lead_magnets'))
+    safe_dir_abs = os.path.abspath(safe_dir)
+    filename = os.path.basename(magnet.file_path)
+    file_path = os.path.join(safe_dir_abs, filename)
+
+    if not os.path.exists(file_path):
+        raise OSError(f"Lead magnet file not found: {file_path}")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)